modules/vault-elb/main.tf

```
@@ -24,6 +24,18 @@ resource "aws_elb" "vault" {
   security_groups = [aws_security_group.vault.id]
   subnets         = var.subnet_ids
 
+  # optional access_logs creation  
+  dynamic "access_logs" {
+    for_each = var.access_logs == null ? [] : ["once"]
+
+    content {
+      enabled       = lookup(access_logs.value, "enabled", lookup(access_logs.value, "bucket", null) != null)
```

NIT: I think defaulting to `null` here would be OK:

```suggestion
      enabled       = lookup(access_logs.value, "enabled", lookup(access_logs.value, "bucket", null))
```

---

Well no right. enabled field takes a bool, so we should assign a bool. enabled = null is not the same as enabled = false. Tbh, i dont think enabled = null will work even.

---

[`enabled`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/elb#enabled) is listed as an optional param, with the default being `true`. If the user doesn't specify it, and we set it to `null`, we get this default. This allows us to fall back to Terraform's defaults, rather than having to provide our own.